docs/diagnostics/PX1015.md

```diff
@@ -0,0 +1,43 @@
+# PX1015
+This document describes the PX1015 diagnostic.
+
+## Summary
+
+| Code   | Short Description                                                                                                                      | Type                           | Code Fix    | 
+| ------ | -------------------------------------------------------------------------------------------------------------------------------------- | ------------------------------ | ----------- | 
+| PX1015 | For a BQL statement that contains parameters, the number of arguments of a `Select` method is different from the number of parameters. | Warning (Level 1: Significant) | Unavailable | 
+
+## Diagnostic Description
+For BQL statements that contain `Required` and `Optional` parameters, a `Select` method must have a number of arguments that is equal to or greater than the number of `Required` parameters in the BQL statement. 
```

Comment:
@k-s-popova this is incorrect - the diagnostic works if the number of arguments passed to the BQL select method doesn't match  the possible number of parameters. For example id the BQL has 1 Required and 2 Optional parameters then the possible number of arguments allowed by the diagnostic is between 1 and 3.
If there are 0 or 4 arguments then the error is displayed.

Reply:
@SENya1990, thanks, fixed.